pkg/R/DataFrame.R

```diff
@@ -0,0 +1,119 @@
+#'              \code{jsonFile}, \code{table} etc.
+#' @rdname DataFrame
+#' @seealso jsonFile, table
+#'
+#' @param env An R environment that stores bookkeeping states of the DataFrame
+#' @param sdf A Java object reference to the backing Scala SchemaRDD
+#' @export
+
+setClass("DataFrame",
+         slots = list(env = "environment",
+                      sdf = "jobj"))
+
+setMethod("initialize", "DataFrame", function(.Object, sdf, isCached, isCheckpointed) {
+  .Object@env <- new.env()
+  .Object@env$isCached <- isCached
+  .Object@env$isCheckpointed <- isCheckpointed
+  
+  .Object@sdf <- sdf
+  .Object
+})
+
+#' @rdname DataFrame
+#' @export
+
+dataFrame <- function(sdf, isCached = FALSE, isCheckpointed = FALSE) {
+  new("DataFrame", sdf, isCached, isCheckpointed)
+}
+
+# The DataFrame accessor function
+
+setGeneric("getsdf", function(df, ...) {standardGeneric("getsdf") })
+setMethod("getsdf", signature(df = "DataFrame" ), function(df) df@sdf )
+
+############################ DataFrame Methods ##############################################
+
+#' Print Schema of a DataFrame
+#' 
+#' Prints out the schema in tree format
+#' 
+#' @param df A SparkSQL DataFrame
+#' 
+#' @rdname printSchema
+#' @export
+
+setGeneric("printSchema", function(df) { standardGeneric("printSchema") })
+
+setMethod("printSchema",
+          signature(df = "DataFrame"),
+          function(df) {
+            sdf <- getsdf(df)
+            schemaString <- callJMethod(sdf, "printSchema")
+            cat(schemaString)
+          })
+
+#' Register Temporary Table
+#' 
+#' Registers a DataFrame as a Temporary Table in the SQLContext
+#' 
+#' @param df A SparkSQL DataFrame
+#' @param tableName A character vector containing the name of the table
+#' 
+#' @rdname registerTempTable
+#' @export
+
+setGeneric("registerTempTable", function(df, tableName) { standardGeneric("registerTempTable") })
+
+setMethod("registerTempTable",
+          signature(df = "DataFrame", tableName = "character"),
+          function(df, tableName) {
+            if (class(df) == "DataFrame") {
+              sdf <- getsdf(df)
+              callJMethod(sdf, "registerTempTable", tableName)    
+            } else {
+              stop("You must specify a DataFrame.")
+            }
+          })
+
+#' Count
+#' 
+#' Returns the number of rows in a DataFrame
+#' 
+#' @param df A SparkSQL DataFrame
+#' 
+#' @rdname count
+#' @export
+
+setMethod("count",
+          signature(x = "DataFrame"),
+          function(x) {
+            sdf <- getsdf(x)
+            callJMethod(sdf, "count")
+          })
+
+#' Collect elements of a DataFrame
+#' 
+#' Returns a list of Row objects from a DataFrame
+#' 
+#' @param df A SparkSQL DataFrame
+#' 
+#' @rdname collect-methods
+#' @export
+
+setMethod("collect",
+          signature(rdd = "DataFrame"),
+          function(rdd){
+            sdf <- getsdf(rdd)
+            list_obj <- callJMethod(sdf, "collect")
```

Comment:
@concretevitamin @shivaram The class name is currently defined as "DataFrame" so that it will stand out like a sore thumb.  This is because between R and SparkSQL, we're talking about data frames in one form or another all over the place, so I used "DataFrame" to specifically refer to the SparkSQL class.

Any input on a better way to handle this?